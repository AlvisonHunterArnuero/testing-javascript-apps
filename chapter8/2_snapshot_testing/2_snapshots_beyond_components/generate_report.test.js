const fs = require("fs");
const { generateReport } = require("./generate_report");

test("generating a .txt report", () => {
  const inventory = [
    { item: "cheesecake", quantity: 8, price: 22 },
    { item: "carrot cake", quantity: 3, price: 18 },
    { item: "macaroon", quantity: 40, price: 6 },
    { item: "chocolate cake", quantity: 12, price: 17 }
  ];

  generateReport(inventory);
  const report = fs.readFileSync("/tmp/report.txt", "utf-8");
  expect(report).toMatchSnapshot();
});
