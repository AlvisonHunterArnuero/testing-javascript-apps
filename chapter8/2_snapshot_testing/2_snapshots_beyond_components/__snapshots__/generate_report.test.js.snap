// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generating a .txt report 1`] = `
"cheesecake - Quantity: 8 - Value: 176
carrot cake - Quantity: 3 - Value: 54
macaroon - Quantity: 40 - Value: 240
chocolate cake - Quantity: 12 - Value: 204
Total value: 63"
`;
